### PR TITLE
Use React.Component instead of React.createClass

### LINKED ISSUE #130: React.createClass is deprecated starting with React v15.5.4
Starting with v15.5.4 React will output warning about `React.createClass` being deprecated. `React.Component` should be used instead. Looks like Sablono already migrated to `React.Component` [here](https://github.com/r0man/sablono/blob/fb5d756c4201598fe8737ae2877e76f9c25a96f1/src/sablono/interpreter.cljc#L34).

### PR DESCRIPTION
**Pull request by roman01la**
Fixes #130 

Tested, seems to work fine.

I had to remove `.isMounted` check [here](https://github.com/tonsky/rum/compare/gh-pages...roman01la:react-15.5.4?expand=1#diff-e43d9dc4d17cf22183180b772a0cf0a8L167) because it is not supported in React classes created with `React.Component`. Not sure if this can potentially break something 🤔 Is it possible that a component will unmount during scheduled update?

### CODE REVIEW

**Comment by roman01la** [comment]
@tonsky Updated with `:rum/mounted?` flag as a replacement for `.isMounted`

File: src/rum/core.cljs [hunk 1]
```diff
@@ -39,12 +39,12 @@
 
         ctor (fn [props]
                (this-as this
-                 (set! (.-state this)
-                       #js {":rum/state"
-                            (-> (gobj/get props ":rum/initial-state")
-                                (assoc :rum/react-component this)
-                                (call-all init props)
-                                volatile!)})
+                 (gobj/set this "state"
```
Comment (sansolovyov) [inline src/rum/core.cljs]: Is that any better than just `set!`? Or `aset`?

Comment (roman01la) [reply src/rum/core.cljs]: I'd say it's "future proof". Also `aset` is going to be deprecated at some point and property name symbols like `-state` in `set!` are mungled by GCC, unless externs provided
```
;; this
(set! this -state true)

;; compiles to this
cljs.user.this$.state = true;
```

**Comment by kurt-o-sys** [comment]
+1

`isMounted` is deprecated and will be removed some day. (It's considered bad practice https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html and well... it's doomed.)
I'm writing RN apps using rum/re-natal and I'm very reluctant of staying with rum because I really think some day, I won't be able to upgrade the app to new versions of RN (and react).

**Comment by Rovanion** [comment]
Is there any chance of this PR being merged soon? Terribly sorry but I'm in the same boat as kurt-o-sys and the warnings are a constant nag.

**Comment by tonsky** [comment]
Yeah I’ll take a look soon

**Comment by tonsky** [comment]
merged with 1390fbc83706a423410b4fbf38bf265762e202ca

**Pull request closed without merging**